services/ros/ros-version.service.js

```diff
@@ -0,0 +1,111 @@
+import { NotFound } from '../index.js'
+import { parseReleaseVersionFromDistro } from './ros-version-helpers.js'
+
+const tagsSchema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      refs: Joi.object({
+        edges: Joi.array()
+          .items({
+            node: Joi.object({
+              name: Joi.string().required(),
+            }).required(),
+          })
+          .required(),
+      }).required(),
+    }).required(),
+  }).required(),
+}).required()
+
+const contentSchema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      object: Joi.object({
+        text: Joi.string().required(),
+      }).allow(null),
+    }).required(),
+  }).required(),
+}).required()
+
+export default class RosVersion extends GithubAuthV4Service {
+  static category = 'version'
+
+  static route = { base: 'ros/v', pattern: ':distro/:packageName' }
+
+  static examples = [
+    {
+      title: 'ROS Package Index',
+      namedParams: { distro: 'humble', packageName: 'vision_msgs' },
+      staticPreview: {
+        ...renderVersionBadge({ version: '4.0.0' }),
+        label: 'humble',
+      },
+    },
+  ]
+
+  async handle({ distro, packageName }) {
+    const tagsJson = await this._requestGraphql({
+      query: gql`
+        query ($refPrefix: String!) {
+          repository(owner: "ros", name: "rosdistro") {
+            refs(
+              refPrefix: $refPrefix
+              first: 30
+              orderBy: { field: TAG_COMMIT_DATE, direction: DESC }
+            ) {
+              edges {
+                node {
+                  name
+                }
+              }
+            }
+          }
+        }
+      `,
+      variables: { refPrefix: `refs/tags/${distro}/` },
+      schema: tagsSchema,
+    })
+
+    // Filter for tags that look like dates: humble/2022-06-10
+    const tags = tagsJson.data.repository.refs.edges
+      .map(edge => edge.node.name)
+      .filter(tag => /^\d+-\d+-\d+$/.test(tag))
+
+    const ref = tags[0] ? `refs/tags/${distro}/${tags[0]}` : 'refs/heads/master'
+    const prettyRef = tags[0] ? `${distro}/${tags[0]}` : 'master'
+
+    const contentJson = await this._requestGraphql({
+      query: gql`
+        query ($expression: String!) {
+          repository(owner: "ros", name: "rosdistro") {
+            object(expression: $expression) {
+              ... on Blob {
+                text
+              }
+            }
+          }
+        }
+      `,
+      variables: {
+        expression: `${ref}:${distro}/distribution.yaml`,
+      },
+      schema: contentSchema,
+    })
+
+    if (!contentJson.data.repository.object) {
+      throw new NotFound({
+        prettyMessage: `distribution.yaml not found: ${distro}@${prettyRef}`,
+      })
+    }
+    const version = parseReleaseVersionFromDistro(
+      contentJson.data.repository.object.text,
+      packageName
+    )
+
+    return { ...renderVersionBadge({ version }), label: distro }
```

Comment:
For version badges, the label is usually the name of the package registry, as opposed to the package.

I think we do want to include ros in the label. With the conda badges, we include the channel in the label separated by a pipe. How about we do similar here - `ros|distro` e.g: ![](https://img.shields.io/badge/ros|humble-1.0.0-blue)

Reply:
Ok, makes sense. What do you think about including spaces around the pipe? Like this: ![](https://img.shields.io/badge/ros%20%7C%20humble-1.0.0-blue)

Reply:
Yeah agreed :+1:  Change it on this PR and I will do another PR to update the conda one.

Reply:
done!